blocks/api/parser.js

```diff
@@ -96,65 +125,17 @@ export function asType( value, type ) {
  * @return {Object}                All block attributes
  */
 export function getBlockAttributes( blockType, rawContent, attributes ) {
-	// Retrieve additional attributes sourced from content
-	const sourcedAttributes = getSourcedAttributes(
-		rawContent,
-		blockType.attributes
-	);
-
-	const blockAttributes = reduce( blockType.attributes, ( result, source, key ) => {
-		let value;
-		if ( sourcedAttributes.hasOwnProperty( key ) ) {
-			value = sourcedAttributes[ key ];
-		} else if ( attributes ) {
-			value = attributes[ key ];
-		}
-
-		// Return default if attribute value not assigned
-		if ( undefined === value ) {
-			// Nest the condition so that constructor coercion never occurs if
-			// value is undefined and block type doesn't specify default value
-			if ( 'default' in source ) {
-				value = source.default;
-			} else {
-				return result;
-			}
-		}
-
-		// Coerce value to specified type
-		const coercedValue = asType( value, source.type );
-
-		if ( 'development' === process.env.NODE_ENV &&
-				! sourcedAttributes.hasOwnProperty( key ) &&
-				value !== coercedValue ) {
-			// Only in case of sourcing attribute from content do we want to
-			// allow coercion, as comment attributes are serialized respecting
-			// original data type. In development environments, log if value
-			// coerced to specified type is not strictly equal. We still allow
-			// coerced value to be assigned into attributes to avoid errors.
-			//
-			// Example:
-			//   Number( 5 ) === 5
-			//   Number( '5' ) !== '5'
-
-			// eslint-disable-next-line no-console
-			console.error(
-				`Expected attribute "${ key }" of type ${ source.type } for ` +
-				`block type "${ blockType.name }" but received ${ typeof value }.`
-			);
-		}
-
-		result[ key ] = coercedValue;
-		return result;
-	}, {} );
+	const blockAttributes = mapValues( blockType.attributes, ( attributeSchema, attributeKey ) => {
+		return getBlockAttribute( attributeKey, attributeSchema, rawContent, attributes );
+	} );
 
 	// If the block supports anchor, parse the id
 	if ( blockType.supportAnchor ) {
 		blockAttributes.anchor = hpqParse( rawContent, attr( '*', 'id' ) );
 	}
 
 	// If the block supports a custom className parse it
-	if ( blockType.className !== false && attributes && attributes.className ) {
+	if ( false !== blockType.className && attributes && attributes.className ) {
```

Comment:
Thinking you might have had to make these changes only because you were running an older version of the branch with newer dependencies? (Related: #3187)

Reply:
>Thinking you might have had to make these changes only because you were running an older version of the branch with newer dependencies? (Related: #3187)

Noting that this received no response. It is not a blocker to me for merge.

Reply:
Oh sorry! missed this one, I think this happened because of my eslint fix on save (I enabled this to compare to prettier). I can revert this change.